### PR TITLE
Local User events

### PR DESCRIPTION
**Pull request by lancewf**
### :nut_and_bolt: Description: What code changed, and why?

Adding create, update, and delete events for Automate local users.

### :chains: Related Resources

https://github.com/chef/automate/pull/4108

### :+1: Definition of Done
Events are created when creating, deleting, and updating Automate's local users. 

### :athletic_shoe: How to Build and Test the Change
1. `build components/automate-gateway && start_all_services`
1. Start the local UI or `rebuild componets/automate-ui`
1. Go to https://a2-dev.test/settings/users and create a new user. 
1. Update that users display name
1. Delete the user. 
1. Go to https://a2-dev.test/dashboards/event-feed and ensure there are three events for each action. 

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
Screen Shot 2020-08-20 at 1 19 10 PM

### CODE REVIEW

File: components/automate-gateway/gateway/services.go [hunk 1]
```diff
@@ -208,7 +208,7 @@ func (s *Server) RegisterGRPCServices(grpcServer *grpc.Server) error {
 	if err != nil {
 		return errors.Wrap(err, "create client for users mgmt service")
 	}
-	pb_iam.RegisterUsersServer(grpcServer, handler_users.NewServer(usersMgmtClient))
+	pb_iam.RegisterUsersServer(grpcServer, handler_users.NewServer(usersMgmtClient, eventFeedClient))
```
Comment (lancewf) [inline components/automate-gateway/gateway/services.go:211]: The event feed client is used to send the event.

File: components/automate-gateway/handler/iam/v2/users/users.go [hunk 2]
```diff
@@ -39,6 +47,11 @@ func (p *Server) CreateUser(
 		return nil, err
 	}
 
+	_, err = p.eventFeedClient.HandleEvent(ctx, createUserCreateEvent(resp))
+	if err != nil {
+		logrus.Errorf("Problem sending event %v", err.Error())
+	}
```
Comment (lancewf) [inline components/automate-gateway/handler/iam/v2/users/users.go:53]: Sending the create user event.

File: components/automate-gateway/handler/iam/v2/users/users.go [hunk 3]
```diff
@@ -63,6 +76,12 @@ func (p *Server) DeleteUser(
 	if err != nil {
 		return nil, err
 	}
+
+	_, err = p.eventFeedClient.HandleEvent(ctx, createUserDeleteEvent(req))
+	if err != nil {
+		logrus.Errorf("Problem sending event %v", err.Error())
+	}
```
Comment (lancewf) [inline components/automate-gateway/handler/iam/v2/users/users.go:83]: Sending the delete user event.

File: components/automate-gateway/handler/iam/v2/users/users.go [hunk 4]
```diff
@@ -89,6 +108,11 @@ func (p *Server) UpdateUser(
 		return nil, err
 	}
 
+	_, err = p.eventFeedClient.HandleEvent(ctx, createUserUpdateEvent(resp))
+	if err != nil {
+		logrus.Errorf("Problem sending event %v", err.Error())
+	}
+
```
Comment (lancewf) [inline components/automate-gateway/handler/iam/v2/users/users.go:115]: Sending the update user event.

File: components/automate-ui/src/app/helpers/event-feed/event-helper.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
+
+import { endsWith, replace } from 'lodash/fp';
```
Comment (lancewf) [inline components/automate-ui/src/app/helpers/event-feed/event-helper.ts:2]: This is a refactor to have the logic of getting the correct icon for an event type and event type display name in one place. This is also adding the new event type of "local_user".

**Review by srenatus — APPROVED** [review]
Code looks good to me.

I'm wondering a little about having local users there, but not local teams.

Also, it may be possible that someone is not allowed to view all users, but see the event feed? They'd then get to know stuff they shouldn't. I'm not certain how realistic this concern is, though.

**Comment by susanev** [comment]
by default, only admins are allowed to see users, and all other users see the event feed so i think the situation @srenatus describes will be common.

**Comment by lancewf** [comment]
I don't see a problem adding events for local teams also. 

@susanev Does that mean we don't want events for local users then?

**Comment by susanev** [comment]
@lancewf 🤷 im not sure, is there anyway to make them only appear if for example the user has permission on `iam:users:list` or similar?

**Comment by lancewf** [comment]
@susanev I don't know an easy way to filter them on those permissions.

**Comment by tylercloke** [comment]
Natalie and the team thinks information around local users should be admin only. Since event feed is all or nothing permissions wise, this is probably not a good thing to add for now if we want that info to be admin only. Long-term, we could consider making event feed function different to filter based on some AuthZ constraints or we could have an admin event feed endpoint that has events like these that we want to be restricted.

**Comment by susanev** [comment]
to clarify one point there, that it should match with the users permissions not necessarily be admin only.

**Comment by lancewf** [comment]
Thank you for considering this. I will close this PR for now.

**Pull request closed without merging**